iothub/device/devdoc/Convention-based operations.md

```diff
@@ -119,38 +119,54 @@ public Task<ClientProperties> GetClientPropertiesAsync(CancellationToken cancell
 /// <param name="propertyCollection">Reported properties to push.</param>
 /// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
 /// <returns>The response containing the operation request Id and updated version no.</returns>
-public Task<ClientPropertiesUpdateResponse> UpdateClientPropertiesAsync(ClientPropertyCollection propertyCollection, CancellationToken cancellationToken = default);
+public Task<ClientPropertiesUpdateResponse> UpdateClientPropertiesAsync(ClientProperties properties, CancellationToken cancellationToken = default);
```

Comment:
we'll need to test how the following flow works:

-> GetClientProperties() -> update some "reported" property values -> Send the same ClientProperties back to service via UpdateClientProperties

Reply:
This is one of the things I don't like about this API.  In this model, a patch is the same object as an entire twin, so we have to expect the user to send in an entire twin when only a patch is expected.  In that case, we can't _really_ honor the contract because the user can send a full twin, but we can only update the reported properties.